spyder/plugins/outlineexplorer/tests/test_widgets.py

```diff
@@ -207,6 +206,52 @@ def test_go_to_cursor_position(create_outlineexplorer, qtbot):
     assert outlineexplorer.treewidget.currentItem().text(0) == 'method1'
 
 
+def test_follow_cursor(create_outlineexplorer, qtbot):
+    """
+    Test that the cursor is followed
+    """
+    outlineexplorer = create_outlineexplorer(TEXT, 'test.py',
+                                             follow_cursor=True)
+    # Move the mouse cursor in the editor to line 31 :
+    editor = outlineexplorer.treewidget.current_editor
+    editor._editor.go_to_line(31)
+    assert editor._editor.get_text_line(31) == "        if False:"
+    # method1 is collapsed
+    assert outlineexplorer.treewidget.currentItem().text(0) == 'test.py'
+
+    # Go to cursor to open the cursor
+    qtbot.mouseClick(outlineexplorer.fromcursor_btn, Qt.LeftButton)
+
+    # Check if follows
+    editor._editor.go_to_line(1)
+    assert outlineexplorer.treewidget.currentItem().text(0) == 'test.py'
+    editor._editor.go_to_line(31)
+    assert outlineexplorer.treewidget.currentItem().text(0) == 'method1'
+
+
+def test_outlineexplorer_updates(create_outlineexplorer, qtbot):
+    """
+    Test that the cursor is followed
+    """
+    outlineexplorer = create_outlineexplorer(TEXT, 'test.py',
+                                             follow_cursor=True)
+    # Move the mouse cursor in the editor to line 5 :
+    editor = outlineexplorer.treewidget.current_editor
+    editor._editor.go_to_line(4)
+    assert editor._editor.get_text_line(4) == ""
+    # Go to cursor to open the cursor
+    qtbot.mouseClick(outlineexplorer.fromcursor_btn, Qt.LeftButton)
+
+    newcell_txt = "# %% newcell"
+    with qtbot.waitSignal(editor.sig_outline_explorer_data_changed):
+        qtbot.keyClicks(editor._editor, newcell_txt)
+    
```

Comment:
This is what ``PEP8Speaks`` is complaining about, FYI. If you turn on real time code style analysis in Spyder, these sorts of errors get highlighted for you automatically.